### DISCUSSION TITLE
How to remove magic strings from AppHost Program.cs

### DISCUSSION THREAD

**Discussion by swegele** [microsoft/aspire#3677] [Q&A] [answered]
Sorry if dumb question, but I couldn't figure it out.  I'm on Aspire preview 5.

**ALL** the examples use magic strings in the apphost and then in the various actuall projects.  
```cs
var myredis = builder.AddRedis("myredis");
var myorleans = builder.AddOrleans("myorleans")
    .WithClustering(myredis )
    .WithGrainStorage("mygrainstore", myredis );
```

I tried creating a constants project which is referenced by XX.AppHost but that doesn't seem to work, probably because it thinks I'm adding it to the orchestration :-)
Any ideas of how to get rid of these magic strings which is use-able from all the projects including XX.AppHost.

**EDIT:** 
A little wonky - but I just tried "Add existing item" on the XX.AppHost project and pointed to the Constants.cs file and used "Add as linked".  
This adds a link to the constants file as if it is in the project.
Seems to compile and work fine so maybe I answered my own question 
:-)

**Comment by davidfowl** [comment, marked as answer]
> I tried creating a constants project which is referenced by XX.AppHost but that doesn't seem to work, probably because it thinks I'm adding it to the orchestration :-)

If you want to do that, then you need to set `IsAspireProjectResource="false"` on your project reference:

https://github.com/dotnet/aspire/blob/f71c3025598e4c428c03cfc701ee4b3fc26884b7/playground/mongo/Mongo.AppHost/Mongo.AppHost.csproj#L17

Shared sources also work.

**Reply by swegele** [reply]
Aha I figured there was a trick!  That will come in handy.  Thanks @davidfowl !

P.S.  My "add existing file as link" trick worked nicely because all the constants can be marked "internal" which adds just a little obfuscation.  Beats sharing a bunch of public const values in an assembly that will get distributed along side projects.
